client-js/app/elements/labrad-plot.ts

```diff
@@ -1126,33 +1127,36 @@ export class Plot extends polymer.Base {
 
 
   /**
+   * Helper function to get mouse position in the coordinates of the svg plot
+   * area. The d3.mouse function returns coordinates relative to the full html
+   * element, so we must account for the margins. We also clip the coordinates
+   * to the available plot area.
+   */
+  private mousePositionClipped(offset: number): number[] {
+    const [x, y] = d3.mouse(this);
+    return [
+      clip(x - this.margin.left, 0, this.width + offset),
+      clip(y - this.margin.top, 0, this.height + offset)
+    ];
+  }
+
+
+  /**
    * Zoom into a selected rectangular region on the graph.
    */
-  private drawZoomRectangle(): void {
+  private drawRectangle(callback: Function): void {
```

Comment:
As an alternate to taking a callback function, you can have this return promise, something like. `Promise<{rect, x1, y1, x2, y2}>`. Then you can use this with async/await for inline code. And you can still call it in callback style if desired by calling `then` on the returned promise.